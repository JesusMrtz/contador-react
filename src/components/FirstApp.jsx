import PropTypes  from 'prop-types';


/**
 * Si no hay una dependencia de comportamiento del componente, las variables y funciones pueden estar fuera
 * Aquellas funciones o variables que estan fuera del Functional Componet, no se vuelve a redibuajar en memoria para mostrarlo.
 * 
 */

const name = 'Jesús Martínez Torres';
const age = 26;
const arrayOfNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const newMessage = {
    author: name,
    message: 'Hola, soy goku'
};

function returnValue() {
    return 'Estoy retornando algo';
}


function FirstApp({ title, propNumber }) {
    return (
        <>
            <h1>{ title }</h1>
            <h2>Estoy recibiendo un prop númerico: { propNumber * 12}</h2>
            <hr />
            <h2>Hola, mi nombre es { name }</h2>
            <h3>{ arrayOfNumbers }</h3>

            {/* Si queremos imprimir un objeto, tenemos que serealizarlo */}
            <p>{ JSON.stringify(newMessage) }</p>
            {/*  */}

            <p>{ returnValue() }</p>

            <h3>Este es un subtitulo del subtitulo { age * 43 }</h3>
        </>
    );
}

/** Agregando tipado a los PropTypes  */
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    propNumber: PropTypes.number
}

/** Implementando defaultProps  */
FirstApp.defaultProps = {
    title: 'No hay titulo',
    // propNumber: 0
}


export default FirstApp;