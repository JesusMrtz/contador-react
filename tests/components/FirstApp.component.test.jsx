import { render } from "@testing-library/react";
import FirstApp from "../../src/components/FirstApp";


describe(`Pruebas en el componente <FirstApp />`, () => {
    it('Debe de hacer match con el snapshot', () => {
        /** 1. Inicialización */
        const title = 'Hola mundo';

        // Renderiza un componente en memoria
        // actualiza el objeto "screen" que es parte del @testing-library
        const { container } = render(<FirstApp title={ title } />);


        /** 3. Aserción */

        /* Se toma una fotografía del componente para evitar los cambios en el document
            * Si un componente cambia constantemente, no es recomendable hacer el snapshot
        */
        expect(container).toMatchSnapshot();
    });


    it('Debe de mostrar la prop "title" en un H1', () => {
        const title = 'Hola soy goku';
        const { container, getByText, getByTestId } = render(<FirstApp  title={ title } />);

        // Verificar en lo que se renderiza exista el titulo "Hola soy goku"
        expect(getByText(title)).toBeTruthy();
        
        // const h1Tag = container.querySelector('h1');

        // // Verificar de que exista un H1
        // expect(h1Tag).toBeTruthy();
        // expect(h1Tag.textContent).toContain(title);


        expect(getByTestId('title').textContent).toContain(title);
    });


    it('Debe de mostrar el "propNumber" enviado por los props', () => {
        const propNumber = 23;
        const { getAllByTestId } = render(<FirstApp propNumber={ propNumber } />);

        expect(getAllByTestId('prop-number').length).toBe(1);
    });
});
