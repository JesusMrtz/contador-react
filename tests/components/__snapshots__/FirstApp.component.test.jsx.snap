// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
    Hola mundo
  </h1>
  <h2
    data-testid="prop-number"
  >
    Estoy recibiendo un prop númerico: 
    NaN
  </h2>
  <hr />
  <h2>
    Hola, mi nombre es 
    Jesús Martínez Torres
  </h2>
  <h3>
    1
    2
    3
    4
    5
    6
    7
    8
    9
  </h3>
  <p>
    {"author":"Jesús Martínez Torres","message":"Hola, soy goku"}
  </p>
  <p>
    Estoy retornando algo
  </p>
  <h3>
    Este es un subtitulo del subtitulo 
    1118
  </h3>
</div>
`;
